src/__tests__/Login.test.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event'
+
+import { Provider } from 'react-redux';
+import store from '../app/store';
+import App from '../App';
+import GlobalContextProvider from '../app/contexts';
+import * as firebaseMock from '../app/firebase-configs';
+
+jest.mock('../features/auth/useRecaptcha', () => jest.fn(() => ({
+  isRecaptchaVerified: true
+})))
+
+jest.useFakeTimers();
+
+describe('/', () => {
+  beforeEach(() => 
+    render(
+        <Provider store={store}>
+          <GlobalContextProvider>
+            <App />
+          </GlobalContextProvider>
+        </Provider>
+      )
+  )
+  
+  it('should login', async () => {
+    const fakeSignIn = jest.spyOn(firebaseMock, 'signInWithEmailAndPassword')

```

Comment:
(PS: you're spying internal code, this does not test the actual firebase integration, this test can still pass even if I uninstall firebase completely from the application. But it's fine by now, it's a good start.)

Reply:
I'm solving this conversation with the caveat that it should be improved in the future.